spec/support/shared_examples_for_api.rb

```diff
@@ -245,8 +245,15 @@
         post_to_validate_endpoint
       end
 
-      it { expect_error_response('london_rates_apply is invalid, london_rates_apply must be true, false or nil') }
-      it { expect(LogStuff).to have_received(:send).with(:error, hash_including(type: 'api-error')) }
+      it 'returns an error' do
+        skip('Only test for LGFS') if claim_user_type == 'Advocate'
+        expect_error_response('london_rates_apply is invalid, london_rates_apply must be true, false or nil')
+      end
```

Comment:
I think I understand why you've done it this way, but I'm not sure I'm completely happy with using `skip` to set the scope of these tests. If we're planning to also add London rates to AGFS claims at some point then _maybe_ it's ok to leave it as it is temporarily, but if this will only ever be for LGFS then I think we should look at structuring the tests differently to avoid the need to use `skip`.

For example you could create a new set of shared examples in this file (in fact two, one for the validate endpoint and one for the create endpoint) just to test the London rates:

```
RSpec.shared_examples 'validate London rates' do
  describe "POST #{ClaimApiEndpoints.for(options[:relative_endpoint]).validate}" do
    subject(:post_to_validate_endpoint) do
      post ClaimApiEndpoints.for(options[:relative_endpoint]).validate, valid_params, format: :json
    end

    before { allow(LogStuff).to receive(:send) }

    let(:claim_user_type) { 'Litigator' }

    <your contexts from these tests>
  end
end
```
and
```
RSpec.shared_examples 'create claim with London rates' do
  describe "POST #{ClaimApiEndpoints.for(options[:relative_endpoint]).create}" do
    subject(:post_to_create_endpoint) do
      post ClaimApiEndpoints.for(options[:relative_endpoint]).create, valid_params, format: :json
    end

    before { allow(LogStuff).to receive(:send) }

    let(:claim_user_type) { 'Litigator' }

   <your contexts from these tests>
end
```
And then you can include them in just the LGFS API tests:
```
include_examples 'validate London rates'
include_examples 'create claim with London rates'
```

(I've not tested any of the above, so it might not actually work quite like that 🙂)

It adds some duplication of setup but more cleanly segregates the tests so that the tests for LGFS functionality only run against LGFS claims, rather than against all claims and we have to skip the ones that are out of scope. This sort of thing is one of the drawbacks of shared contexts. They can become overcomplicated surprisingly quickly.

Reply:
Thanks Mark, I understand what you've said here. I'll check with Angela and Steffan whether we'd expect AGFS to eventually have this functionality and then we can chat about the best way to neaten this up? If it going to be LGFS only I like your solution above 👍

Reply:
I've had a talk with Angela and Steffan. There are no unimplemented fees in our backlog for AGFS which would make use of the london_rates_apply flag, and while we can't see the future, we've no particular reason to expect any will be added in the near future.  I'll have a go at structuring the tests into some new shared example sets like you've suggested above